app/src/main/java/fr/neamar/kiss/searcher/HistorySearcher.java

```diff
@@ -22,21 +22,28 @@ public HistorySearcher(MainActivity activity) {
     }
 
     @Override
-    protected List<Pojo> doInBackground(Void... voids) {
+    protected Void doInBackground( Void... voids )
+    {
         // Ask for records
         boolean smartHistory = !prefs.getString("history-mode", "recency").equals("recency");
         boolean excludeFavorites = prefs.getBoolean("exclude-favorites", false);
-        
+
         // Convert `"number-of-display-elements"` to double first before truncating to int to avoid
         // `java.lang.NumberFormatException` crashes for values larger than `Integer.MAX_VALUE`
         int maxRecords = (Double.valueOf(prefs.getString("number-of-display-elements", String.valueOf(DEFAULT_MAX_RESULTS)))).intValue();
 
+        MainActivity activity = activityWeakReference.get();
+        if( activity == null )
+            return null;
+
         //Gather favorites
-        ArrayList<Pojo> favoritesPojo = new ArrayList<Pojo>(0);
+        ArrayList<Pojo> favoritesPojo = new ArrayList<>( 0 );
         if(excludeFavorites){
             favoritesPojo = KissApplication.getDataHandler(activity).getFavorites(activity.tryToRetrieve);
         }
 
-        return KissApplication.getDataHandler(activity).getHistory(activity, maxRecords, smartHistory, favoritesPojo);
+        List<Pojo> pojos = KissApplication.getDataHandler(activity).getHistory(activity, maxRecords, smartHistory, favoritesPojo);
+        this.addResult( pojos.toArray(new Pojo[0]) );
```

Comment:
That's the second time you use this syntax, I'm unclear what it means--a constructor with array notation?